src/c_sharp_graph/query.rs

```diff
@@ -8,174 +9,232 @@ use regex::Regex;
-        let search: Search = self.get_search(query)?;
-
-        debug!("search: {:?}", search);
+pub enum QueryType<'graph> {
+    All {
+        db: &'graph StackGraph,
+        source_type: &'graph SourceType,
+    },
+    Method {
+        db: &'graph StackGraph,
+        source_type: &'graph SourceType,
+    },
+}
 
-        let mut results: Vec<ResultNode> = vec![];
+#[derive(Eq, Hash, PartialEq, Debug)]
+pub(crate) struct FQDN {
+    pub(crate) namespace: Option<String>,
+    pub(crate) class: Option<String>,
+    pub(crate) method: Option<String>,
+}
 
-        // If we are search for all things from a ref
-        // ex: System.Configuration.ConfigurationManager.* or System.Configuration.*
-        // this means that we need to find the Nodes from the namespace, then find all the matches
-        // for all the nodes in that namespace.
-        if search.all_references_search() {
-            // get all the compilation units that use some portion of the search (using System or
-            // using System.Configuration) This will require us to then determine if there qualified
-            // names ConfigurationManager.AppSettings for examples;
-
-            // We will also need to find the definition of this by looking at the namespace
-            // declaration. then we need to capture all the nodes that are definitions attached to
-            // this (for instance namespace System.Configuration; Class ConfigurationManager; method
-            // AppSettings)
-            let mut definition_root_nodes: Vec<Handle<Node>> = vec![];
-            let mut referenced_files: HashSet<Handle<File>> = HashSet::new();
-            let mut file_to_compunit_handle: HashMap<Handle<File>, Handle<Node>> = HashMap::new();
-
-            for node_handle in self.db.iter_nodes() {
-                let node: &Node = &self.db[node_handle];
-                let file_handle = match node.file() {
-                    Some(h) => h,
-                    None => {
-                        continue;
-                    }
-                };
-                let symbol_option = node.symbol();
-                if symbol_option.is_none() {
-                    // If the node doesn't have a symbol to look at, then we should continue and it
-                    // only used to tie together other nodes.
-                    continue;
+pub(crate) fn get_fqdn(node: Handle<Node>, graph: &StackGraph) -> Option<FQDN> {
+    let mut fqdn = FQDN {
+        namespace: None,
+        class: None,
+        method: None,
+    };
+    // traverse upwards based on the FQDN edge
+    // Once there is no FQDN edge, return
+    let n = &graph[node];
+    let source_info = graph
+        .source_info(node)
+        .expect("FQDN nodes must have source info");
+    let syntax_type = &graph[source_info
+        .syntax_type
+        .into_option()
+        .expect("FQDN nodes must have a syntax type")];
```

Comment:
NOTE: this needs to not be expect and give back a reasonable error to handle

This may mean trying to re-load the graph for this particular file, or something after giving an internal error back to the user.

Reply:
Will do this is a follow up #33